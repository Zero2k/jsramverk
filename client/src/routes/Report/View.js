import React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import Button from '@material-ui/core/Button';

const ViewReport = ({ report, remove }) => {
  return (
    <div>
      <h2>{report.title}</h2>
      <div dangerouslySetInnerHTML={{ __html: report.text }} />
      <Button
        color="primary"
        variant="outlined"
        component={RouterLink}
        to={`/reports/edit/${report.id}`}
      >
        Edit
      </Button>
      <Button
        style={{ marginLeft: '10px' }}
        color="primary"
        variant="outlined"
        onClick={() => remove(report)}
      >
        Delete
      </Button>
    </div>
  );
};

export default ViewReport;
