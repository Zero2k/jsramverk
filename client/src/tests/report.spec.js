import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount } from 'enzyme';

import View from '../routes/Report/View';

configure({ adapter: new Adapter() });

const report = { title: 'title', text: 'text' };

describe('Report View', () => {
  it('should render correctly', () => {
    const component = shallow(<View report={report} />);
  
    expect(component).toMatchSnapshot();
  });

  it('should have a title', () => {
    const component = mount(
      <MemoryRouter>
        <View report={report} />
      </MemoryRouter>
    );
    
    const title = component.find('h2');

    expect(title.text()).toBe('title');
  });
});
