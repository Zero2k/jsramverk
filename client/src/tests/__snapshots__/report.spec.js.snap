// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report View should render correctly 1`] = `ShallowWrapper {}`;
